yarn-project/simulator/src/public/public_processor_metrics.ts

```diff
@@ -54,6 +61,27 @@ export class PublicProcessorMetrics {
       description: 'Size of deployed bytecode',
       unit: 'By',
     });
+
+    this.totalGas = meter.createGauge(Metrics.PUBLIC_PROCESSOR_TOTAL_GAS, {
+      description: 'Total gas used in block',
+      unit: 'gas',
+    });
+
+    this.totalGasHistogram = meter.createHistogram(Metrics.PUBLIC_PROCESSOR_TOTAL_GAS_HISTOGRAM, {
+      description: 'Total gas used in block as histogram',
+      unit: 'gas',
```

Comment:
Just a heads up that the default histogram buckets only go up to 10k

https://github.com/open-telemetry/opentelemetry-js/blob/8dc74e609a992ba6379c99033244e46ed902dc04/packages/sdk-metrics/src/view/Aggregation.ts#L87-L90

Reply:
Sorted, thanks